core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -298,7 +276,8 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, time: Time, met
     partitionStateMachine.startup()
 
     // register the partition change listeners for all existing topics on failover
-    controllerContext.allTopics.foreach(topic => registerPartitionModificationsListener(topic))
+    registerPartitionModificationsHandlers(controllerContext.allTopics.toSeq)
+    kafkaControllerZkUtils.registerZNodeChangeHandlers(partitionModificationsHandlers.values.toSeq)
```

Comment:
This line seems unnecessary since it's done in the previous line.

Reply:
Good catch. This must've been some leftover code while refactoring.